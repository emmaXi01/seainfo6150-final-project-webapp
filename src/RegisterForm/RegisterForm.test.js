import React from "react";
import { render } from "@testing-library/react";
import RegisterForm from "./RegisterForm.jsx";

describe("RegisterForm tests", () => {
  const course = {
    "slug": "181122102925-statue-of-liberty-crown",
    "category": "Computer Science",
    "title": "Java Programming Masterclass for Software Developers",
    "shortText": "Learn Java In This Course And Become a Computer Programmer. Obtain valuable Core Java Skills And Java Certification",
    "instructor": "Kate Farley",
    "pubDate": "11/2020",
    "image": {
      "_url": "https://img-a.udemycdn.com/course/240x135/533682_c10c_4.jpg?nZ46z0qISmfa8qZFWvm7OpzmPyGFyNfEOoSySIzYsx6snPkgqmsyFUGUkjWkr84nTmwc5wuy5d2Bsla_fyNXS0Ct9txcLvpUI9erTWZ0CNQULjdN6KrFhZhg72uK",
      "_height": "619",
      "_width": "1100"
    },
    "skillsLearning": "Learn the core Java skills needed to apply for Java developer positions in just 14 hours.",
    "content": ["Java, a High Level Language with a Purpose.", "How Java Programming Works."],
    "feedBack": {
        "reviewer":"Rafael Santos",
        "text":"I'm really learning a LOT. To be honest some topics are extensive, and sometimes too heavy on detail (and a bit boring at first), but if you insist, Tim always finds a way to make it interesting again. "
    }
  }

  it("renders the register form component correctly", () => {
    const { container } = render(<RegisterForm onBack={()=>{}} course={course}/>);
    expect(container).toMatchSnapshot();
  });
});

