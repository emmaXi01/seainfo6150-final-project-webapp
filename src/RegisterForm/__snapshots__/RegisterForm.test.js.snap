// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm tests renders the register form component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="topContainer"
    >
      <div
        class="topWrapper"
      >
        <h1>
          Java Programming Masterclass for Software Developers
        </h1>
        <p>
          Learn Java In This Course And Become a Computer Programmer. Obtain valuable Core Java Skills And Java Certification
        </p>
      </div>
    </div>
    <form
      class="form"
    >
      <fieldset>
        <legend>
          <h2>
            Registeration Form
          </h2>
        </legend>
        <div>
          <label
            for="first-name"
          >
            First name
            <sup>
              *
            </sup>
            :
            <input
              class="name"
              id="first-name"
              name="first-name"
              required=""
              type="text"
            />
          </label>
          <label
            for="last-name"
          >
            Last name
            <sup>
              *
            </sup>
            :
            <input
              class="name"
              id="last-name"
              name="last-name"
              required=""
              type="text"
            />
          </label>
        </div>
        <div>
          <label
            for="email"
          >
            Email
            <sup>
              *
            </sup>
            :
            <input
              class="email"
              id="email"
              name="email"
              placeholder="ex: myname@example.com"
              required=""
              type="email"
            />
          </label>
        </div>
        <div>
          <label
            for="telephone"
          >
            Telephone:
            <input
              class="phone"
              id="telephone"
              name="telephone"
              pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
              type="tel"
            />
            <small>
              format: 123-456-7890
            </small>
          </label>
        </div>
        <div>
          <label
            for="education"
          >
            Education Level:
            <select
              class="education"
              id="education"
            >
              <option>
                please select one
              </option>
              <option
                value="Doctoral Degree"
              >
                Doctoral Degree
              </option>
              <option
                value="Master's Degree"
              >
                Master's Degree
              </option>
              <option
                value="Bachelor's Degree"
              >
                Bachelor's Degree
              </option>
              <option
                value="High School Diploma"
              >
                High School Diploma
              </option>
              <option
                value="Less Than a High School"
              >
                Less Than a High School
              </option>
            </select>
          </label>
          <label
            for="college"
          >
            School/College:
            <input
              class="college"
              id="college"
              name="tcollege"
              type="text"
            />
          </label>
        </div>
        <div>
          <label>
            Certificate Type:
          </label>
          <input
            class="certificate"
            name="digit"
            type="radio"
            value="digit"
          />
          <label>
            digit
          </label>
          <input
            class="certificate"
            name="paper"
            type="radio"
            value="paper"
          />
          <label>
            paper
          </label>
          <input
            class="certificate"
            name="both"
            type="radio"
            value="both digit and paper"
          />
          <label>
            both digit and paper
          </label>
        </div>
        <div>
          <label
            for="reason"
          >
            Why do you want to register:
            <select
              class="reason"
              id="reason"
            >
              <option
                value="true"
              >
                please select one
              </option>
              <option
                value="Interesting"
              >
                Interesting
              </option>
              <option
                value="Get a course certificate"
              >
                Get a course certification
              </option>
              <option
                value="Find a job"
              >
                Find a job
              </option>
            </select>
          </label>
        </div>
        <div>
          <input
            class="registerButton"
            type="submit"
            value="Register"
          />
          <button
            class="cancelButton"
          >
            Cancel
          </button>
        </div>
      </fieldset>
    </form>
  </div>
</div>
`;
